sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Connection } from "typeorm";
+import {
+  DesignationAgreement,
+  DesignationAgreementLocation,
+  DesignationAgreementStatus,
+  Institution,
+  InstitutionLocation,
+  User,
+} from "../../database/entities";
+
+/**
+ * Manages the operations needed for designation agreements that are submitted by the institutions
+ * and then need to be reviewed by the Ministry for approval or denial. While the Ministry is
+ * approving the designation it can also edit dates and locations as needed.
+ */
+@Injectable()
+export class DesignationAgreementService extends RecordDataModelService<DesignationAgreement> {
+  constructor(connection: Connection) {
+    super(connection.getRepository(DesignationAgreement));
+  }
+
+  /**
+   * Initiates a new designation agreement request. This action
+   * is meant to me initiated by the institution signing officer
+   * for further assessment of the Ministry.
+   * The designation agreement and the location that are part of it will
+   * be saved as part of the same DB transaction.
+   * @param institutionId institution id requesting the designation.
+   * @param submittedData dynamic data that represents the designation requested.
+   * @param submittedByUserId institution user submitting the designation.
+   * @param submittedDate date that the institution user submitted the designation.
+   * @param requestedLocationsIds ids of the locations requested to be designated.
+   * @returns the new designation agreement created.
+   */
+  async submitDesignationAgreement(
+    institutionId: number,
+    submittedData: any,
```

Comment:
when I see any, I assume it is because this is a part of PR. Please correct me If I am wrong.

Reply:
This is not because it is part of the PR. Part of the reason that the Ministry is using form.io is because there are some forms that are flexible in a way that they could be changed in many ways without impacting the application or process at all. The area of the form here defined as "any" does not have a fixed structure.

The data is going to be the result of an area in the form.io that is not required to have any constraints and could be freely changed by the Ministry. The data under this particular section is also not supposed to be used anywhere in the API, only to populate the same form again for visualization, so, I do not see a reason for this particular case.